skipper_test.go

```diff
@@ -384,6 +385,268 @@ func Example_ratelimitRegistryBinding() {
+		DataClients:     []routing.DataClient{dc},
+		PostProcessors: []routing.PostProcessor{
+			loadbalancer.NewAlgorithmProvider(),
+			reg,
+		},
+		SuppressLogs: true,
+	}
+	rt := routing.New(ro)
+	defer rt.Close()
+	<-rt.FirstLoad()
+	tracer := &tracingtest.Tracer{}
+	pr := proxy.WithParams(proxy.Params{
+		Routing:     rt,
+		OpenTracing: &proxy.OpenTracingParams{Tracer: tracer},
+	})
+	defer pr.Close()
+	lb := stdlibhttptest.NewServer(pr)
+	defer lb.Close()
+
+	rsvo := routesrv.Options{
+		Address:                         ":8082",
+		KubernetesURL:                   lb.URL,
+		KubernetesRedisServiceNamespace: "skipper",
+		KubernetesRedisServiceName:      "redis",
+		KubernetesHealthcheck:           true,
+		SourcePollTimeout:               1500 * time.Millisecond,
+	}
+
+	go routesrv.Run(rsvo)
+
+	ready := false
+	for !ready {
+		rsp, _ := http.DefaultClient.Head("http://localhost:8082/routes")
+		if rsp != nil && rsp.StatusCode == 200 {
+			ready = true
+		}
+		time.Sleep(100 * time.Millisecond)
+	}
+
+	// run skipper proxy that we want to test
+	o := Options{
+		Address:                        ":9090",
+		EnableRatelimiters:             true,
+		EnableSwarm:                    true,
+		Kubernetes:                     true,
+		SwarmRedisRemote:               "http://localhost:8082/swarm/redis/shards",
+		KubernetesURL:                  lb.URL,
+		KubernetesHealthcheck:          true,
+		SourcePollTimeout:              1500 * time.Millisecond,
+		WaitFirstRouteLoad:             true,
+		ClusterRatelimitMaxGroupShards: 2,
+		SwarmRedisDialTimeout:          100 * time.Millisecond,
+		SuppressRouteUpdateLogs:        false,
+		SupportListener:                ":9091",
+		testOptions: testOptions{
+			redisUpdateInterval: time.Second,
+		},
+	}
+
+	sigs := make(chan os.Signal, 1)
+	go run(o, sigs, nil)
+
+	// wait for proxy being ready
+	ready = false
+	for !ready {
+		rsp, _ := http.DefaultClient.Get("http://localhost:9090/kube-system/healthz")
+		if rsp != nil && rsp.StatusCode == 200 {
+			ready = true
+		}
+		time.Sleep(100 * time.Millisecond)
+	}
+
+	rate := 10
+	sec := 5
+	va := httptest.NewVegetaAttacker("http://localhost:9090/test", rate, time.Second, time.Second)
+	va.Attack(io.Discard, time.Duration(sec)*time.Second, "mytest")
+	t.Logf("Success [0..1]: %0.2f", va.Success())
+
+	if successRate := va.Success(); successRate < 0.1 && successRate > 0.5 {
+		t.Fatalf("Test should have a success rate between %0.2f < %0.2f < %0.2f", 0.1, successRate, 0.5)
+	}
+	if reqCount := va.TotalRequests(); reqCount < uint64(rate*sec) {
+		t.Fatalf("Test should run %d requests got: %d", uint64(rate*sec), reqCount)
+	}
+	countOK, ok := va.CountStatus(http.StatusOK)
+	if countOK == 0 {
+		t.Fatalf("Some requests should have passed: %d %v", countOK, ok)
+	}
+
+	countLimited, ok := va.CountStatus(http.StatusTooManyRequests)
+	if !ok || countLimited < countOK {
+		t.Fatalf("count TooMany should be higher than OKs: %d < %d: %v", countLimited, countOK, ok)
+	}
```

Comment:
you have `clusterRatelimit("foo", 1, "1s")`  so you should check that you allow 5*1rps (+/-1 rps) and rest  StatusTooManyRequests